backend/amd64/selection.ml

```diff
@@ -274,10 +274,21 @@ method! select_operation op args dbg =
          assert false
     end
   | Cextcall { func = "caml_int64_bits_of_float_unboxed"; alloc = false;
-               ty = [|Int|]; ty_args = [XFloat] }
+               ty = [|Int|]; ty_args = [XFloat] } ->
+      (match args with
+      | [Cop(Cload ((Word_int | Word_val | Double), _), [loc], _dbg)] ->
```

Comment:
Are `Word_int` and `Word_val` possible here?

Reply:
I don't think they are currently possible, but I thought it may be useful to accept here (and in the symmetric case) "chunks" of the same width, because it is safe.

Reply:
Given the title of the pull request, I can see the
reason behind accepting `Word_int`. However,
if memory serves, `Word_val` is either a naked
pointer or a tagger integer. Reinterpreting such
a value as a naked integer, while not necessarily
unreasonable, does not appear to be in line with
the contract of `caml_int64_bits_of_float_unboxed`.

Reply:
Fixed.